openedx/core/djangoapps/catalog/tests/test_utils.py

```diff
@@ -152,90 +163,185 @@ def test_munge_catalog_program(self):
         self.assertEqual(munged, expected)
 
 
-@mock.patch(UTILS_MODULE + '.get_edx_api_data')
-@mock.patch('config_models.models.cache.get', return_value=None)
-class TestGetCourseRun(mixins.CatalogIntegrationMixin, TestCase):
-    """Tests covering retrieval of course runs from the catalog service."""
+@httpretty.activate
+class TestGetCourseRun(mixins.CatalogIntegrationMixin, CacheIsolationTestCase):
+    """
+    Tests covering retrieval of course runs from the catalog service.
+    """
+
+    ENABLED_CACHES = ['default']
+
     def setUp(self):
         super(TestGetCourseRun, self).setUp()
 
         self.user = UserFactory()
-        self.course_key = CourseKey.from_string('foo/bar/baz')
         self.catalog_integration = self.create_catalog_integration()
+        self.course_key_1 = 'foo1/bar1/baz1'
+        self.course_key_2 = 'foo2/bar2/baz2'
+        self.course_key_3 = 'foo3/bar3/baz3'
+        self.course_key_4 = 'foo4/bar4/baz4'
```

Comment:
Don't do this. The catalog factories at https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/catalog/tests/factories.py will create all of this for you. For example, look at how we create test data in https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/catalog/tests/test_utils.py#L62. Follow the same pattern in your tests here; let the factories worry about creating the test data.